### PR TITLE
Add windows bootstrap agent image

### PR DESCRIPTION
**Pull request by orangedeng**
refer to https://github.com/rancher/rancher/issues/10448
and https://github.com/rancher/rancher/issues/10442

Add a new image to bootstrap windows host agent service.

### CODE REVIEW

**Comment by cjellick** [comment]
win-agent should not be a top level directory in the project. should probaby be under the package directory and that directory (and the scripts that interact with it) possibly restructured to have linux and windows subdirs

**Comment by cjellick** [comment]
what builds this image? If it is an image that we are going to release, it should be built by the ci scripts. I believe the `scripts/package` script handles building the linux image, so you can start there for a reference point.

**Comment by cjellick** [comment]
On second thought, I'm ok if building this image is not integrated into the ci scripts, but it should still be moved into a subdir under the package directory. Also, can you update the project's README.md with instructions on how to build this bootstrap image?

File: win-agent/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM microsoft/nanoserver:10.0.14393.1593
+ENV RANCHER_AGENT_WINDOWS_IMAGE rancher/win-agent:v0.0.1
+RUN mkdir "c:/program files/rancher"
+VOLUME [ "c:/program files/rancher" ]
+ADD [ "http://dev-tool.oss-cn-shenzhen.aliyuncs.com/windows/devcon.exe","c:/program files/" ]
```
Comment (cjellick) [inline win-agent/Dockerfile]: It seems bad to be downloading third party binaries. What does this binary do? Could we at least do an md5sum check on the binary? If we do that, you might have to move downloading the binary and md5suming it to the build-image script.

Comment (orangedeng) [reply win-agent/Dockerfile]: That is a tool for handling device on windows. There is no way I can download from Microsoft alone. So I put it into cloud storage.

**Comment by orangedeng** [comment]
@cjellick 
I can not find any package script in `./scripts` dir. Do I miss anything?

If I add a package dir and put win-agent into it, should I move ./agent into it?

I can write a new README.md for how to build the windows agent image, and I would like to put it into win-agent.

**Comment by cjellick** [comment]
To be honest, I was thinking that this was in the rancher/agent repository bc we moved the scripts for building the linux image there for 2.0.
Apologies for the confusion.
Now I think where you have it is ok.

File: agent/win-agent/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM microsoft/nanoserver:10.0.14393.1593
+ENV RANCHER_AGENT_WINDOWS_IMAGE rancher/win-agent:v0.0.1
```
Comment (deniseschannon) [inline agent/win-agent/Dockerfile]: we're going to name this `rancher/agent-windows`

**Comment by cjellick** [comment]
Please change dir to agent-windows instead of win-agent

File: agent/win-agent/build-image.ps1 [hunk 3]
```diff
@@ -0,0 +1,21 @@
+$IMAGE=$(get-item env:IMAGE -ErrorAction Ignore)
+
+Invoke-WebRequest -Uri "http://dev-tool.oss-cn-shenzhen.aliyuncs.com/windows/devcon.exe" -OutFile devcon.exe
```
Comment (deniseschannon) [inline agent/win-agent/build-image.ps1]: Is it possible to get this hosted on github/rancher versus aliyun?

Comment (orangedeng) [reply agent/win-agent/build-image.ps1]: Is it ok that I just put it under `./agent-windows`? And we don't event need to download it or validate it.

**Comment by cjellick** [comment]
LGTM

**Pull request merged**